canonical-web-and-design/maas-ui
# feat(ui): add workloads warning + erase checkboxes to machine release form

## Done

- Added `ReleaseForm` component for use in machine list and machine details, which adds:
  - Checkboxes for erasing disks on release
  - A warning if one or more of the machines have defined workload annotations

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open the release form for a machine without workload annotations.
- Check that the release form shows and it has checkboxes for erasing disks. Check the "Enable erasing" checkbox.
- Release the machine and check that it goes through a "Disk erasing" status.
- Open the release form for a machine with workload annotations (this [old PR](https://github.com/canonical-web-and-design/maas-ui/pull/2014) has details for adding annotations).
- Check that the release form shows a warning.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2334

## Screenshot
![Screenshot_2021-02-11 Machines bolla MAAS](https://user-images.githubusercontent.com/25733845/107601988-f3aa5f80-6c73-11eb-9cec-43df7d5c7164.png)
